code/controllers/subsystem/thermal_dissipation.dm

```diff
@@ -9,35 +9,172 @@ var/list/thermal_dissipation_atoms = list()
 	display_order = SS_DISPLAY_THERM_DISS
 
 	var/list/currentrun
-	var/atom/A //The atom currently being processed by the subsystem.
+
+	//Keep all of these here to avoid having to redeclare them for every datum/reagents on every tick.
+	var/datum/reagents/R
+	var/datum/gas_mixture/the_air
+	var/atom/A
+	var/turf/T
+	var/emission_factor
+	var/atom/this_potentially_insulative_layer
+	var/i
```

Comment:
if this is a for loop iterator pre declaring it feels idiotic
surely byond isn't THAT shit

t. not an expert

Reply:
I'm not sure either, but moving all those out there does seem to make things faster.
I'm thinking BYOND can probably figure out
```
var/i
for(i in ...)
```
and
`for (var/i in)`

But in this case it would still have to redeclare the variable once for every reagents datum being processed, every tick, since it's a for loop inside in a proc being called inside a while loop.

Reply:
Have you actually tested this in particular? Yes, you have to redeclare the variables every time if they're local, but it's also faster to access local variables than object variables.
Also, this is very ugly and the difference is almost certainly minuscule

Reply:
Thanks for testing. I'll change that.